app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/ChangePasswordScreen.kt

```diff
@@ -0,0 +1,198 @@
+package com.simply.birthdayapp.presentation.ui.screens.main.profile
+
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.simply.birthdayapp.R
+import com.simply.birthdayapp.presentation.ui.components.AppBaseTopBar
+import com.simply.birthdayapp.presentation.ui.components.AuthButton
+import com.simply.birthdayapp.presentation.ui.components.PasswordTextFiled
+import com.simply.birthdayapp.presentation.ui.theme.AppTheme
+import org.koin.androidx.compose.getViewModel
+
+@Composable
+fun ChangePasswordScreen(
+    profileViewModel: ProfileViewModel,
+    onSignOutClicked: () -> Unit = {},
+    navToProfileScreen: () -> Unit = {},
+) {
+    val oldPassword by profileViewModel.oldPassword.collectAsState()
+    val password by profileViewModel.password.collectAsState()
+    val repeatPassword by profileViewModel.repeatPassword.collectAsState()
+    val hasPasswordError by profileViewModel.hasPasswordError.collectAsState()
+    val hasOldPasswordError by profileViewModel.hasOldPasswordError.collectAsState()
+    val hasRepeatPasswordError by profileViewModel.hasRepeatPasswordError.collectAsState()
+    val doneButtonEnabled by profileViewModel.enableDoneButton.collectAsState()
+    val changePasswordSuccess by profileViewModel.changePasswordSuccess.collectAsState()
+    val changePasswordErrorMessage by profileViewModel.changePasswordErrorMessage.collectAsState()
+    val changePasswordErrorState by profileViewModel.changePasswordErrorState.collectAsState()
+    val context = LocalContext.current
+
+    LaunchedEffect(changePasswordSuccess) {
+        if (changePasswordSuccess) {
+            Toast.makeText(context, (R.string.change_password_success), Toast.LENGTH_SHORT).show()
```

Comment:
No need to have String ResID in brackets

Reply:
Fixed